journal.py

```diff
@@ -139,40 +144,37 @@ def logout(request):
 def add_entry(request):
     if request.authenticated_userid:
         try:
-            write_entry(request)
-        except psycopg2.Error:
-            # This will catch any errors generated by the database
+            Entry.from_request(request)
+        except SQLAlchemyError:
             return HTTPInternalServerError
-        cursor = request.db.cursor()
-        cursor.execute(NEWEST_ENTRY)
-        keys = ('id', 'title', 'text', 'created')
-        entry = dict(zip(keys, cursor.fetchone()))
-        entry['text'] = markdown.markdown(entry['text'], extensions=['codehilite', 'fenced_code'])
-        entry['created'] = entry['created'].strftime('%b. %d, %Y')
-        return entry
+        new = Entry.recent()
+        json_add = {
+            'id': new.id,
+            'title': new.title,
+            'text': markdown.markdown(new.text,
+                                      extensions=['codehilite', 'fenced_code']),
+            'created': new.created.strftime('%b. %d, %Y')
+        }
+        return json_add
```

Comment:
Seems to me like a chunk of stuff like this ought to be a method on the entry object.  Something like `entry.as_json()`.  To preserve sanity, you could even give it an optional `display` argument, which if True would return a dict that had the markdown rendered text, but if false would have the plain text version.